### PR TITLE
New taiga.auth module

### PR DESCRIPTION
**Pull request by bameda**
> NOTE: Use https://github.com/kaleidos-ventures/taiga-front/pull/68 to run the frontend. 
> NOTE (2): Related to https://github.com/kaleidos-ventures/taiga-doc/pull/10

This PR overrides the auth module in order to make the authentication system more secure.  The code is based on [djangorestframework-simplejwt](https://github.com/jazzband/djangorestframework-simplejwt) adapting it so that the api and the interfaces were 100% compatible with the old Taiga authentication module. 

You can customize the settings for user tests, especially lifetimes and periodicities, with

```python
SIMPLE_JWT = {                                                                                                                                                                
    'ACCESS_TOKEN_LIFETIME': timedelta(minutes=1),
    'REFRESH_TOKEN_LIFETIME': timedelta(minutes=3),
    'FLUSH_REFRESHED_TOKENS_PERIODICITY': 3 * 60, # secons
}
```

### HOW TO TESTS:

There are a lot of changes in the code, the best approach may be to organize a peer review meeting and I can give you "a walk" through the new code. 

- [x] - Review test and code (we can have a meet to a pair review).
- [x] - Test login 
    - [x] - with user/pass
    - [x] - with some contrib plugin (github or gitlab)
    - [x] - seed
- [x] Test register a new user
- [x] Test project invitations
    - [x] For a new user (register)
    - [x] For existing user (login)  
- [x] - Test expiration after an inactivity period (remember, after refresh toke has expired)

### CODE REVIEW

File: settings/common.py [hunk 1]
```diff
@@ -439,6 +442,14 @@
 
 
 AUTH_USER_MODEL = "users.User"
+
+SIMPLE_JWT = {
+    'ACCESS_TOKEN_LIFETIME': timedelta(hours=1),
+    'REFRESH_TOKEN_LIFETIME': timedelta(days=3),
+    'CANCEL_TOKEN_LIFETIME': timedelta(days=100),
+    'FLUSH_REFRESHED_TOKENS_PERIODICITY': 3 * 24 * 3600, # secons
```
Comment (yamila-moreno) [inline settings/common.py]: typo > "seconds"

File: taiga/auth/settings.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+  SIMPLE_JWT = {
+      'ACCESS_TOKEN_LIFETIME': timedelta(minutes=5),
+      'CANCEL_TOKEN_LIFETIME': timedelta(days=30),
+      'REFRESH_TOKEN_LIFETIME': timedelta(days=1),
+      'ROTATE_REFRESH_TOKENS': False,
+      'BLACKLIST_AFTER_ROTATION': True,
+      'UPDATE_LAST_LOGIN': False,
+
+      'FLUSH_REFRESHED_TOKENS_PERIODICITY': 3 * 24 * 3600, # secons
```
Comment (yamila-moreno) [inline taiga/auth/settings.py]: typo > "seconds"

File: taiga/auth/settings.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+DEFAULTS = {
+    'ACCESS_TOKEN_LIFETIME': timedelta(hours=60),
+    'REFRESH_TOKEN_LIFETIME': timedelta(days=3),
+    'CANCEL_TOKEN_LIFETIME': timedelta(days=100),
+    'ROTATE_REFRESH_TOKENS': True,
+    'BLACKLIST_AFTER_ROTATION': True,
+    'UPDATE_LAST_LOGIN': True,
+
+    'FLUSH_REFRESHED_TOKENS_PERIODICITY': 3 * 24 * 3600, # secons
```
Comment (yamila-moreno) [inline taiga/auth/settings.py]: typo > "seconds"

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 ## 6.3.0 (unreleased)
 
+- New Auth module, based on djangorestframework-simplejwt (history #tg-4625, issue #tgg-626))
```
Comment (yamila-moreno) [inline CHANGELOG.md:5]: typos
- history > story
- tgg-626 > tg-626

File: taiga/auth/compat.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    def __init__(self, value):
+        self.value = value
+
+    def __bool__(self):
+        return self.value
+
+    def __call__(self):
+        warnings.warn(
+            "Using user.is_authenticated() and user.is_anonymous() as a method "
+            "is deprecated. Remove the parentheses to use it as an attribute.",
```
Comment (yamila-moreno) [inline taiga/auth/compat.py:59]: typo parenthesis

**Review by yamila-moreno — DISMISSED** [review]
This review includes some typos.

**Review by tdelatorre — APPROVED** [review]

**Review by yamila-moreno — APPROVED** [review]

**Pull request merged**